src/middlewared/middlewared/plugins/network.py

```diff
@@ -489,8 +491,11 @@ class DNSService(Service):
                 nameservers.append(gc['gc_nameserver3'])
 
         resolvconf = ''
-        if domain:
-            resolvconf += 'search {}\n'.format(domain)
+
+        if domains:
+            for domain in domains:
+                resolvconf += 'search {}'.format(domain)
+
```

Comment:
You will end up with several seach entries, which seems wrong.

You want "search" followed by all domains, space delimited.